custom/icds_reports/static/js/directives/download/download.directive.js

```diff
@@ -319,11 +339,31 @@ function DownloadController($scope, $rootScope, $location, locationHierarchy, lo
                 }
             });
             vm.selectedMonth = vm.selectedMonth >= 7 ? vm.selectedMonth : 7;
+        } else if((year.id === 2019 && vm.isPPRSelected())) {
+            var currentMonth = latest.getMonth() + 1;
+            var currentYear = latest.getFullYear();
+            vm.months = _.filter(vm.monthsCopy, function (month) {
+                if (currentYear === 2019) {
+                    return month.id >= 4 && month.id <= currentMonth;
+                } else {
+                    return month.id >= 4;
+                }
+            });
+            vm.selectedMonth = vm.selectedMonth >= 4 ? vm.selectedMonth : 4;
+            vm.quarters = vm.quartersCopy.slice(1,4);
+            vm.selectedQuarter = vm.selectedQuarter >= 2 ? vm.selectedQuarter : 2;
```

Comment:
I feel the same, like this file (download.directive.js) is a bit messy since we are encountering every new logic for every new report (but some refactoring we could do with util function). Need to plan out first.